cybench/datasets/modified_dataset.py

```diff
@@ -0,0 +1,32 @@
+import pandas as pd
+
+from cybench.datasets.dataset import Dataset
+
+
+class ModifiedTargetsDataset(Dataset):
+    def __init__(
+        self,
+        dataset: Dataset,
+        modified_targets: pd.DataFrame = None,
+    ):
+        """
+        Dataset wrapper to support modified target values.
+
+        :param dataset: Dataset with original targets
+        :param modified_targets: pandas.DataFrame that contains modified targets
+        """
+        self._crop = dataset.crop
+        self._df_y = modified_targets.sort_index()
```

Comment:
I find this a bit tricky; is df_x also sorted with `sort_index()`?
The current check is only whether the datasets contain the same indices; are entries of dataset always retrieved by their index, or also by their row_number or so?

Reply:
get_item() in the Dataset class does the following:
`data_x = self._get_feature_data(loc_id, year)` so ths mitigates the risk that df_x and df_y get out of sync.

Reply:
> I find this a bit tricky; is df_x also sorted with `sort_index()`? The current check is only whether the datasets contain the same indices; are entries of dataset always retrieved by their index, or also by their row_number or so?

Yes data is sorted. See Dataset(). We only retrieve inputs by keys from targets not by row numbers.